webapp/src/main/webapp/src/app/groups/groups.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from "@angular/core";
+import { GroupFilterOptions } from "./model/group-filter-options.model";
+import { ActivatedRoute } from "@angular/router";
+import { GroupService } from "./groups.service";
+import { GroupQuery } from "./model/group-query.model";
+import { Group } from "./model/group.model";
+
+@Component({
+  selector: 'groups',
+  templateUrl: './groups.component.html'
+})
+export class GroupsComponent implements OnInit {
+  filterOptions: GroupFilterOptions;
+  groups: Group[];
+  query: GroupQuery;
+
+  constructor(private route: ActivatedRoute, private service: GroupService) {
+  }
+
+  ngOnInit() {
+    this.filterOptions = this.route.snapshot.data[ "filterOptions" ];
+
+    this.query = new GroupQuery(this.filterOptions.subjects);
+
+    this.query.school = this.filterOptions.schools[ 0 ];
+    this.query.schoolYear = this.filterOptions.schoolYears[ 0 ];
+    this.query.subject = this.filterOptions.subjects[ 0 ];
```

Comment:
Not sure what the defaults should be here so just selecting the first in each array.  I probably need to do some null / array checking before doing this.